grpc-sys/src/lib.rs

```diff
@@ -97,33 +169,83 @@ impl From<i32> for GrpcStatusCode {
     }
 }
 
+/// Result of a gRPC call.
+///
+/// If the caller satisfies the prerequisites of a
+/// particular operation, the `GrpcCallStatus` returned will be `Ok`.
+/// Receiving any other value listed here is an indication of a bug in the caller.
+///
+/// Based on `grpc_call_error`.
 #[repr(C)]
 #[derive(Debug, PartialEq)]
 pub enum GrpcCallStatus {
+    /// Everything went ok.
     Ok = 0,
+
+    /// Something failed, we don't know what.
     Error,
+
+    /// This method is not available on the server.
     ErrorNotOnServer,
+
+    /// This method is not available on the client.
     ErrorNotOnClient,
+
+    /// This method must be called before server_accept.
```

Comment:
You may wish to fence `server_accept`, as well as `writes_done` and `write_status`. Not a blocker though.

Reply:
This is intentionally kept identical to gRPC's doc.